### PR TITLE
 Update Style Manager dialog with new item types 

### LINKED ISSUE #3364: [FEATURE] Double clicking a style .xml in the browser now opens
the manager dialog, allowing browsing and non-edit actions for the
style.
Original commit: https://github.com/qgis/QGIS/commit/701a29fd6ed7a505a6601ad38105a0c8bc712b41 by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #3365: [FEATURE] When browsing a style from the browser panel, allow copying
symbols to the default style from the right click menu and share button
Original commit: https://github.com/qgis/QGIS/commit/a1b44c939d474279a7e901a924c577fde7d6dc1a by nyalldawson

Allows users to easily transfer symbols and ramps from a style xml
file directly to their local style database.

### LINKED ISSUE #3857: [FEATURE] Allow creation and editing of text formats through style manager dialog
Original commit: https://github.com/qgis/QGIS/commit/1e718478df2366b48c75ef1b81aa85669f4caeb2 by nyalldawson

This allows the style manager to be used to manage text formats,
which store the font, color, buffers, shadows, and backgrounds
of text formats (i.e. all the formatting parts of label settings,
which can be used outside of labels, eg in layouts)

Text Formats are added alongside the existing symbols and color
ramp management options, and offer identical functionality as these.

### LINKED ISSUE #3860: [FEATURE] Allow storage of label settings in style manager
Original commit: https://github.com/qgis/QGIS/commit/0fd4984b3ba14ab068185fc31b83798c3bb9a49b by nyalldawson

Just like the earlier change which allows for text formats to
be managed in style manager, this change allows for layer-wide
label settings to be stored and managed in styles.

So while a text format includes just the font settings and other
appearance related settings, Label Settings also includes layer-type
specific settings such as label placement, priority, rendering
settings.

### PR DESCRIPTION
**Pull request by DelazJ**
### Description
* Add text format (fixes #3857) and label settings (fixes #3860) as new style types
* Update the main dialog screenshot
* Replace many occurrences of "symbols and color ramps" text with a more generic "style items" expression and adjust description
* Update and harmonize style manager screenshots using the same environment (Ubuntu) for all the screenshots, with more 3.x-like style items
* Mention import styes from browser panel (fixes #3364 and fixes #3365)

PS: Since section titles can no longer be bound to "symbols and color ramps" and have (to be) changed, Help buttons in the related dialogs will fail to open the right section for QGIS>=3.6. So...

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/style_library.rst [hunk 1]
```diff
@@ -21,22 +21,22 @@
 The Style Manager
 ==================
 
-The :guilabel:`Style Manager` is the place where users can manage and create
-generic symbols and color ramps to be used in several QGIS projects. You can
-open that modeless dialog:
+The :guilabel:`Style Manager` is the place where you can manage and create
+generic style items. These are symbols, color ramps, text formats or label
+settings that can be used to symbolize features, layers or print layouts.
+They are stored in the QGIS installation style database and shared with all
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/style_library.rst]: @nyalldawson (sorry to so often ping you but you deserve it... :smiley: ) would you mind reviewing this, please? Not sure where the styles are actually stored.

Comment (nyalldawson) [reply source/docs/user_manual/working_with_vector/style_library.rst]: It's under the user profile folder, `symbology-style.db`. So it's a per-profile file.

File: source/docs/user_manual/working_with_vector/style_library.rst [hunk 2]
```diff
@@ -152,47 +165,49 @@ You can export a set of items to an :file:`.XML` file:
...
-#. In the new dialog, indicate the source of the symbols (it can be an
+#. In the new dialog, indicate the source of the style items (it can be an
    :file:`.xml` file on the disk or a url).
 #. Set whether to |unchecked| :guilabel:`Add to favorites` the items to import.
 #. Check |unchecked| :guilabel:`Do not import embedded tags` to avoid the import
    of tags associated to the items being imported.
 #. Give the name of any :guilabel:`Additional tag(s)` to apply to the new items.
 #. Select from the preview the symbols you want to add to your library.
-#. And press **Import**.
+#. And press :guilabel:`Import`.
```
Comment (nyalldawson) [inline source/docs/user_manual/working_with_vector/style_library.rst]: Might be worth mentioning that you can also browser to exported style .xml files in the browser panel and double click them to explore them and import items from them.

File: source/docs/user_manual/working_with_vector/style_library.rst (hunk 1 above)
Comment (DelazJ) [reply source/docs/user_manual/working_with_vector/style_library.rst]: Thanks. Will update accordingly.

File: source/docs/user_manual/working_with_vector/style_library.rst (hunk 2 above)
Comment (DelazJ) [reply source/docs/user_manual/working_with_vector/style_library.rst]: Yes, found the issue report when looking for the ones potentially fixed to mention in this PR message so it was too late. Will add as a note in a later step.

**Pull request merged**